types/taiko/index.d.ts

```diff
@@ -0,0 +1,366 @@
+    export function client(): any; // TODO: no TS Bindings available: https://github.com/cyrus-and/chrome-remote-interface/issues/112
+    // https://taiko.gauge.org/#switchto
+    export function switchTo(targetUrl: string): Promise<void>;
+    // https://taiko.gauge.org/#intercept
+    // https://github.com/getgauge/taiko/issues/98#issuecomment-42024186
+    export function intercept(requestUrl: string, options?: InterceptMockData | interceptRequestHandler | InterceptRedirectUrl): Promise<void>;
+    // https://taiko.gauge.org/#emulatenetwork
+    export function emulateNetwork(
+        networkType: 'GPRS' | 'Regular2G' | 'Good2G' | 'Good3G' | 'Regular3G' | 'Regular4G' | 'DSL' | 'WiFi, Offline'
+    ): Promise<void>;
+    // https://taiko.gauge.org/#emulatedevice
+    export function emulateDevice(deviceModel: string);
+    // https://taiko.gauge.org/#setviewport
+    export function setViewPort(options: ViewPortOptions): Promise<void>;
+    // https://taiko.gauge.org/#opentab
+    export function openTab(targetUrl: string, options?: NavigationOptions): Promise<void>;
+    // https://taiko.gauge.org/#closetab
+    export function closeTab(targetUrl: string): Promise<void>;
+    // https://taiko.gauge.org/#overridepermissions
+    export function overridePermissions(origin: string, permissions: string[]): Promise<void>;
+    // https://taiko.gauge.org/#clearpermissionoverrides
+    export function clearPermissionOverrides(): Promise<void>;
+    // https://taiko.gauge.org/#setcookie
+    export function setCookie(name: string, value: string, options?: CookieDetailOptions): Promise<void>;
+    // https://taiko.gauge.org/#deletecookies
+    export function deleteCookies(cookieName?: string, options?: CookieOptions): Promise<void>;
+    // https://taiko.gauge.org/#getcookies
+    export function getCookies(options?: { urls: string[] }): Promise<Object[]>;
+    // https://taiko.gauge.org/#setlocation
+    export function setLocation(options: LocationOptions): Promise<void>;
+
+    /**
+     * Page Actions
+     */
+
+    // https://taiko.gauge.org/#goto
+    export function goto(url: string, options?: NavigationOptions): Promise<void>;
+    // https://taiko.gauge.org/#reload
+    export function reload(url: string, options?: NavigationOptions): Promise<void>;
+    // https://taiko.gauge.org/#goback
+    export function goBack(options?: NavigationOptions): Promise<void>;
+    // https://taiko.gauge.org/#goforward
+    export function goForward(options?: NavigationOptions): Promise<void>;
+    // https://taiko.gauge.org/#title
+    export function title(): Promise<string>;
+    // https://taiko.gauge.org/#click
+    export function click(selector: SearchElement | MouseCoordinates, options?: ClickOptions, args?: string[]): Promise<void>;
+    // https://taiko.gauge.org/#doubleclick
+    export function doubleClick(selector: SearchElement | MouseCoordinates, options?: BasicNavigationOptions, args?: string[]): Promise<void>;
```

Comment:
args is a rest parameter and its type has to be RelativeSearchElement

Reply:
changed